client/src/components/Form/Elements/FormText.vue

```diff
@@ -0,0 +1,138 @@
+<template>
+    <div>
+        <b-alert v-if="errorMessage" class="mt-2" :show="dismissCountDown" variant="info" @dismissed="resetAlert">
+            {{ errorMessage }}
+        </b-alert>
+        <b-row align-v="center">
+            <b-col>
+                <b-form-textarea
+                    v-if="inputArea"
+                    :id="id"
+                    v-model="currentValue"
+                    class="text-input"
+                    :readonly="readonly"
+                    :placeholder="placeholder"
+                    :style="style"
+                    @change="onInputChange">
+                </b-form-textarea>
+                <b-form-input
+                    v-else
+                    :id="id"
+                    v-model="currentValue"
+                    class="text-input"
+                    :readonly="readonly"
+                    :placeholder="placeholder"
+                    :style="style"
+                    :type="acceptedTypes"
+                    :list="`${id}-datalist`"
+                    @change="onInputChange">
+                </b-form-input>
+                <datalist v-if="datalist && !inputArea" :id="`${id}-datalist`">
+                    <option v-for="data in datalist" :key="data.value" :label="data.label" :value="data.value"></option>
+                </datalist>
+            </b-col>
+        </b-row>
+    </div>
+</template>
+
+<script>
+export default {
+    props: {
+        value: {
+            // String; Array for multiple
+            default: "",
+        },
+        id: {
+            type: String,
+            default: "",
+        },
+        type: {
+            type: String,
+            default: "text",
+        },
+        area: {
+            // <textarea> instead of <input> element
+            type: Boolean,
+            default: false,
+        },
+        multiple: {
+            // Allow multiple entries to be created
+            type: Boolean,
+            default: false,
+        },
+        readonly: {
+            type: Boolean,
+            default: false,
+        },
+        placeholder: {
+            type: String,
+            default: "",
+        },
+        color: {
+            type: String,
+            default: "",
+        },
+        styleObj: {
```

Comment:
This prop does not exist in current dev, and does not seem to be used anywhere. Please correct me if I am missing something here.

Reply:
I think I added this to allow consumers to pass style objects the conventional Js way (as a nicety) but if it' doesn't seem useful please feel free to remove. I think that would mean that no `style` can ever be passed (only `color`)?
> Apply custom styling to the input element with prop `style={ attr: value, ... }`

Reply:
It's not that it's not useful, it was just not being used or called. Then again, it's better to keep the functionality specific, so atomically just changing the `color` makes it easier for us to test this feature and to ensure that it works well with Galaxy's UI theme, at the same time it covers the previous functionality.